src/fastga_he/models/propulsion/components/connectors/h2_fuel_system/components/sizing_h2_fuel_system_length.py

```diff
@@ -0,0 +1,155 @@
+            self.declare_partials(of="*", wrt="*", val=1.0)
+        elif position == "in_the_middle":
+            self.add_input("data:geometry:cabin:length", val=np.nan, units="m")
+            self.declare_partials(of="*", wrt="*", val=1.0)
+        else:
+            self.add_input("data:geometry:cabin:length", val=np.nan, units="m")
+
+            if wing_related:
+                self.add_input("data:geometry:wing:span", val=np.nan, units="m")
+                self.add_input(
+                    "data:propulsion:he_power_train:H2_fuel_system:"
+                    + h2_fuel_system_id
+                    + ":CG:y_ratio",
+                    val=0.5,
+                    desc="Y position of the power source center of gravity as a ratio of the wing "
+                    "half-span",
+                )
+                self.declare_partials("*", "*", method="exact")
+
+            self.declare_partials("*", "data:geometry:cabin:length", val=0.5)
+
+        self.add_output(
+            "data:propulsion:he_power_train:H2_fuel_system:"
+            + h2_fuel_system_id
+            + ":dimension:length",
+            units="m",
+            val=5.0,
+            desc="Total length of the h2 fuel system",
+        )
+
+    def compute(self, inputs, outputs, discrete_inputs=None, discrete_outputs=None):
+        h2_fuel_system_id = self.options["h2_fuel_system_id"]
+        wing_related = self.options["wing_related"]
+        position = self.options["position"]
+        compact = self.options["compact"]
+
+        if compact:
+            outputs[
+                "data:propulsion:he_power_train:H2_fuel_system:"
+                + h2_fuel_system_id
+                + ":dimension:length"
+            ] = inputs["data:geometry:wing:MAC:length"]
+
+        elif position == "in_the_middle":
+            outputs[
+                "data:propulsion:he_power_train:H2_fuel_system:"
+                + h2_fuel_system_id
+                + ":dimension:length"
+            ] = inputs["data:geometry:cabin:length"]
+        else:
+            l_in_fus = 0.5 * inputs["data:geometry:cabin:length"]
+            if wing_related:
+                outputs[
+                    "data:propulsion:he_power_train:H2_fuel_system:"
+                    + h2_fuel_system_id
+                    + ":dimension:length"
+                ] = (
+                    l_in_fus
+                    + 0.5
+                    * inputs["data:geometry:wing:span"]
+                    * inputs[
+                        "data:propulsion:he_power_train:H2_fuel_system:"
+                        + h2_fuel_system_id
+                        + ":CG:y_ratio"
+                    ]
+                )
+            else:
+                outputs[
+                    "data:propulsion:he_power_train:H2_fuel_system:"
+                    + h2_fuel_system_id
+                    + ":dimension:length"
+                ] = l_in_fus
```

Comment:
Just a suggestion, I don't know if it is better
```suggestion
            if wing_related:
                l_in_wing = 0.5
                    * inputs["data:geometry:wing:span"]
                    * inputs[
                        "data:propulsion:he_power_train:H2_fuel_system:"
                        + h2_fuel_system_id
                        + ":CG:y_ratio"
                    ]
            else:
                 l_in_wing = 0.
            outputs[
                "data:propulsion:he_power_train:H2_fuel_system:"
                + h2_fuel_system_id
                + ":dimension:length"
            ] = l_in_fus + l_in_wing
```
And the indent might be a bit wrong, be careful to check it